File: content/examples/other/leetcode/problems/contains-duplicate/example-2/Solution.java
import java.util.*;

public class Solution {
  public static boolean containsDuplicate(int[] nums) {
    HashSet<Integer> s = new HashSet<>();
    for (int x : nums) {
      if (s.contains(x)) return true;
      s.add(x);
    }
    return false;
  }

  public static void main(String[] args) {
    System.out.println(containsDuplicate(new int[]{1,2,3,1}));
  }
}

File: content/examples/other/leetcode/problems/contains-duplicate/example-2/solution.js
function containsDuplicate(nums) {
  const s = new Set();
  for (const x of nums) {
    if (s.has(x)) return true;
    s.add(x);
  }
  return false;
}

if (require.main===module) {
  console.log(containsDuplicate([1,2,3,1]));
}

module.exports = { containsDuplicate };
